dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -223,215 +140,131 @@ private static Link getLinkFields(Map<String, String> map) throws ParseException
+        if (results != null && !results.isEmpty()) {
+            for (Map<String, Object> map : results) {
                 ret.add(getFolderFields(map));
             }
         }
         return ret;
     }
 
     @NotNull
-    private static Folder getFolderFields(Map<String, String> map) throws ParseException {
+    private static Folder getFolderFields(Map<String, Object> map) throws ParseException {
         Folder folder;
         folder = new Folder();
-        folder.setInode(map.get(INODE));
-        folder.setOwner(map.get(OWNER));
-
-        if (map.get(IDATE) != null && !map.get(IDATE).isEmpty()) {
-            folder.setIDate(map.get(IDATE));
-        }
-
-
-        folder.setName(map.get("name"));
-        folder.setTitle(map.get(TITLE));
-
-        if (map.get(SHOW_ON_MENU) != null && !map.get(SHOW_ON_MENU).isEmpty()) {
-            folder.setShowOnMenu(Boolean.parseBoolean(map.get(SHOW_ON_MENU)));
-        }
-
-        if (map.get(SORT_ORDER) != null && !map.get(SORT_ORDER).isEmpty()){
-            folder.setSortOrder(Integer.parseInt(map.get(SORT_ORDER)));
-        }
-
-        folder.setFilesMasks(map.get("files_masks"));
-
-        folder.setIdentifier(map.get(IDENTIFIER));
-
-        folder.setDefaultFileType(map.get("default_file_type"));
-
-        if (map.get(MOD_DATE) != null && !map.get(MOD_DATE).isEmpty()){
-            folder.setModDate(df.parse(map.get(MOD_DATE)));
-        }
+        folder.setInode((String) map.get(INODE));
+        folder.setOwner((String) map.get(OWNER));
+        folder.setIDate((Date) map.get(IDATE));
+        folder.setName((String) map.get("name"));
+        folder.setTitle((String) map.get(TITLE));
+        folder.setShowOnMenu((Boolean) map.getOrDefault(SHOW_ON_MENU,false));
+        folder.setSortOrder((Integer) map.getOrDefault(SORT_ORDER,0));
+        folder.setFilesMasks((String) map.get("files_masks"));
+        folder.setIdentifier((String) map.get(IDENTIFIER));
+        folder.setDefaultFileType((String) map.get("default_file_type"));
+        folder.setModDate((Date) map.get(MOD_DATE));
         return folder;
     }
 
-    public static List<Container> convertDotConnectMapToContainer(final List<Map<String,String>> results)
-            throws ParseException {
+    public static List<Container> convertDotConnectMapToContainer(final List<Map<String, Object>> results) throws ParseException {
 
         List<Container> ret = new ArrayList<>();
 
-        if(results != null && !results.isEmpty()){
-            for (Map<String, String> map : results) {
+        if (results != null && !results.isEmpty()) {
+            for (Map<String, Object> map : results) {
                 ret.add(getContainerFields(map));
             }
         }
         return ret;
     }
 
     @NotNull
-    private static Container getContainerFields(Map<String, String> map) throws ParseException {
+    private static Container getContainerFields(Map<String, Object> map) throws ParseException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.text.ParseException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)